cdap-app-fabric/src/main/java/io/cdap/cdap/internal/events/SparkProgramStatusMetricsProvider.java

```diff
@@ -61,53 +64,75 @@ public class SparkProgramStatusMetricsProvider implements MetricsProvider {
   private final CConfiguration cConf;
   private final HttpRequestConfig httpRequestConfig;
   private final Integer maxTerminationMinutes;
+  private MetricsCollectionService metricsCollectionService;
 
   @Inject
-  public SparkProgramStatusMetricsProvider(CConfiguration cConf) {
+  public SparkProgramStatusMetricsProvider(CConfiguration cConf, MetricsCollectionService metricsCollectionService) {
     int connectionTimeoutMs = cConf.getInt(Constants.HTTP_CLIENT_CONNECTION_TIMEOUT_MS);
     int readTimeoutMs = cConf.getInt(Constants.HTTP_CLIENT_READ_TIMEOUT_MS);
 
     this.httpRequestConfig = new HttpRequestConfig(connectionTimeoutMs, readTimeoutMs, false);
     this.maxTerminationMinutes = cConf.getInt(Constants.Spark.SPARK_METRICS_PROVIDER_MAX_TERMINATION_MINUTES, 5);
     this.cConf = cConf;
+    this.metricsCollectionService = metricsCollectionService;
   }
 
   @Override
   public ExecutionMetrics[] retrieveMetrics(ProgramRunId runId) {
     if (!runId.getType().equals(ProgramType.SPARK)) {
       return new ExecutionMetrics[]{ExecutionMetrics.emptyMetrics()};
     }
+    long startTime = System.currentTimeMillis();
```

Comment:
This might capture complete failures at high latencies due to the retry logic right?

Reply:
Yes, there is a success tag that can be used to group the success cases and failure case latencies.